src/core/coap/coap_secure.cpp

```diff
@@ -158,6 +158,13 @@ otError CoapSecure::SendMessage(Message &aMessage, otCoapResponseHandler aHandle
 
     VerifyOrExit(IsConnected(), error = OT_ERROR_INVALID_STATE);
 
+#if OPENTHREAD_CONFIG_COAP_BLOCKWISE_TRANSFER_ENABLE
+    if (CoapBase::GetMaxBlockSize() > OT_COAP_OPTION_BLOCK_LENGTH_512)
```

Comment:
Why are we limiting the block size to 512 bytes?  1024 byte blocks do work. :-)

Reply:
That is actually a leftover from a test i've run, thanks, I've corrected it back to 1024 byte block-size.